app/models/backend/transaction.rb

```diff
@@ -1,102 +1,43 @@
+# frozen_string_literal: true
+
+# Used by Stats tests and backend random data generator
 module Backend
-  class Transaction # this stuff is used by the stats tests, beware
-    attr_accessor :estimated_usage
-    attr_accessor :id
-    attr_accessor :provider_account_id
-    attr_accessor :created_at
-    #TODO: there's some duplication between this and delegate :to => :cinstance_data
-    attr_accessor :service_id
-    attr_accessor :client_ip
-    attr_accessor :cinstance_id
-    attr_accessor :log          # hash with the log info
+  class Transaction
+    attr_accessor :id, :estimated_usage, :provider_account_id, :created_at, :service_id, :client_ip, :cinstance_id, :log
 
     attr_writer :user_key
-    attr_writer :confirmed
 
-    delegate :provider_verification_key, :plan_name, :application_id,
-             :service_id, :to => :cinstance_data
+    delegate :provider_verification_key, :plan_name, :application_id, to: :cinstance_data
 
     def initialize(attributes = {})
-      @confirmed = false
```

Comment:
All usages of this class are with `confirmed: true`, and I really-really doubt we want to use it with `confirmed: false`, this functionality seems really strange and useless.

Reply:
What is an unconfirmed transaction?

Reply:
So, this seems to be some really old code, from the times where the `/transactions` endpoints (aka Service Management API) was implemented in System (or so it seems).

Looks like there was a possibility to send transactions, but not confirm them - which means that they would just be saved in storage (Redis), and only after confirming them, they would increment the aggregated usage (and so then deleted from that storage).

This is related: https://github.com/3scale/porta/pull/3647/files#r1412106695